### PR TITLE
feat: support extcontact

### PR DESCRIPTION
**Pull request by frontwindysky**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

### CODE REVIEW

File: lib/api/extcontact.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+   * }
+   */
+  * create(params) {
+    assert(params.name, 'params.name required');
+    assert(params.mobile, 'params.mobile required');
+    assert(params.label_ids, 'params.label_ids required Array<Number>');
+    assert(params.follower_user_id, 'params.follower_user_id required');
+
+    params = Object.assign({}, {
+      state_code: '86',
```
Comment (dead-horse) [inline lib/api/extcontact.js]: params = Object.assign( {state_code: '86' }, params);

**Comment by dead-horse** [comment]
@atian25 看一下，没问题帮忙发个版本，我没权限

File: lib/api/client.js [hunk 2]
```diff
@@ -213,6 +213,8 @@ module.exports = class Client {
       accessToken = yield this.getAccessToken();
     }
     const url = `${this.options.host}/${api}?access_token=${encodeURIComponent(accessToken)}`;
+    console.log(url);
```
Comment (atian25) [inline lib/api/client.js]: remove

File: lib/api/extcontact.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+      size: 100,
+    };
+
+    do {
+      params.offset = params.offset + (100 * page);
+      const users = yield this.list(params);
+      count = users.length;
+      page += 1;
+      all = all.concat(users);
+    } while (count === 100);
```
Comment (atian25) [inline lib/api/extcontact.js]: 这里用 https://github.com/sindresorhus/p-all ？

**Comment by atian25** [comment]
@dead-horse 你回头合并发吧，给你权限了

File: lib/api/client.js (hunk 2 above)
Comment (frontwindysky) [reply lib/api/client.js]: 嗯，漏掉了

File: lib/api/extcontact.js (hunk 3 above)
Comment (frontwindysky) [reply lib/api/extcontact.js]: 这里每次依赖前一次的结果数量决定是否需要再次发起请求，没法做成并发吧？

**Pull request merged**